dask/dataframe/io/parquet/arrow.py

```diff
@@ -1279,36 +1272,19 @@ def _update_metadata_options(
         )
 
     @classmethod
-    def _process_metadata(
+    def _organize_row_groups(
```

Comment:
The `_process_metadata` logic is not being changed (for pyarrow engines) in this PR.  The changes in this file all correpond to re-organization of the same code.

**Reason**:  The `_process_metadata` method is now used by both the "pyarrow" and "fastparquet" engines, and its logic is now separated into: (1) a `_organize_row_groups` **class** method and (2) a `_row_groups_to_parts` utility.  Note that the `_row_groups_to_parts` is located in `utils.py`, because it performs the same high-level logic for all engines.  Any engine-specific logic to be executed by `_row_groups_to_parts` must be embedded in the `make_part_func` arg and the `make_part_kwargs` kwarg.